rclpy/rclpy/handle.py

```diff
@@ -44,7 +45,10 @@ def __init__(self, pycapsule):
         self.__use_count = 0
         self.__request_invalidation = False
         self.__valid = True
-        self.__lock = Lock()
+        self.__lock = RLock()
```

Comment:
Maybe renaming to `__rlock` so when reading code using it it is clear that it is a recursive lock?

Reply:
Renamed `__lock` -> `__rlock` in 8c95ea6